### PR TITLE
feat(datastore): SUM and AVG aggregations

### PR DESCRIPTION
**Pull request by bhshkh**

### CODE REVIEW

**Comment by bhshkh** [comment]
DO NOT MERGE until feature is GA

File: datastore/testdata/index.yaml [hunk 1]
```diff
@@ -38,4 +38,17 @@ indexes:
...
+  properties:
+  - name: T
+  - name: I
+  - name: V
+
+- kind: SQChild
+  ancestor: yes
+  properties:
+  - name: T
+  - name: W
```
Comment (enocom) [inline datastore/testdata/index.yaml:54]: nit missing newline at end of file.

File: datastore/integration_test.go [hunk 2]
```diff
@@ -739,20 +742,75 @@ func TestIntegration_AggregationQueries(t *testing.T) {
...
 			wantErrMsg:  "",
 			wantAggResult: map[string]interface{}{
 				"count": &pb.Value{ValueType: &pb.Value_IntegerValue{IntegerValue: 5}},
 			},
 		},
+		{
+			desc:        "Multiple aggregations",
+			aggQuery:    NewQuery("SQChild").Ancestor(parent).Filter("T=", now).NewAggregationQuery().WithSum("I", "i_sum").WithAvg("I", "avg").WithSum("V", "v_sum"),
+			wantFailure: false,
+			wantErrMsg:  "",
```
Comment (enocom) [inline datastore/integration_test.go:763]: style nit: would it be better to omit this and rely on the zero value?

**Review by enocom — APPROVED** [review]

File: datastore/testdata/index.yaml (hunk 1 above)
Comment (bhshkh) [reply datastore/testdata/index.yaml]: Done

File: datastore/integration_test.go (hunk 2 above)
Comment (bhshkh) [reply datastore/integration_test.go]: Done

File: datastore/integration_test.go [hunk 3]
```diff
@@ -701,17 +704,17 @@ func TestIntegration_AggregationQueries(t *testing.T) {
 	client := newTestClient(ctx, t)
 	defer client.Close()
 
-	parent := NameKey("SQParent", "TestIntegration_Filters"+suffix, nil)
+	parent := NameKey("SQParent", "TestIntegration_AggregationQueries"+suffix, nil)
```
[the remark this answers is not in the document]
Comment (bhshkh) [reply datastore/integration_test.go]: Done

[the remark this answers is not in the document]
Comment (bhshkh) [reply datastore/integration_test.go]: No testcases are being removed here. The data for this testcase TestIntegration_AggregationQueries was copied from Filters testcase (https://github.com/googleapis/google-cloud-go/blob/main/datastore/integration_test.go#L533-L551) but the name wasn't modified by mistake. So, modified now.

**Review by codyoss — APPROVED** [review]
LGTM

**Pull request merged**